ako-infra/ingestion/vcf_k8s_controller.go

```diff
@@ -134,87 +136,89 @@ func (c *VCFK8sController) AddNCPSecretEventHandler(k8sinfo K8sinformers, stopCh
 	}
 }
 
-func (c *VCFK8sController) AddNSXNetworkConfigEventHandler(k8sinfo K8sinformers, stopCh <-chan struct{}) {
-	// Saves the initial nsx network config count during reboot,
+func (c *VCFK8sController) AddNamespaceEventHandler(k8sinfo K8sinformers, stopCh <-chan struct{}) {
+	// Saves the initial workload namespace count during reboot,
 	// before the config handlers are started.
-	if err := addNSXNetworkConfigCount(); err != nil {
-		utils.AviLog.Fatalf("Unable to list NSXNetworkConfig: %s", err.Error())
+	if err := c.addWorkloadNamespaceCount(); err != nil {
+		utils.AviLog.Fatalf("Unable to list Namespaces: %s", err.Error())
 	}
 
-	nsxNetworkConfigHandler := cache.ResourceEventHandlerFuncs{
+	namespaceHandler := cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			utils.AviLog.Infof("NSX Network Config ADD Event")
-			handleNSXNetworkConfigAdd()
+			utils.AviLog.Infof("Namespace ADD Event")
+			c.handleNamespaceAdd()
 		},
 		UpdateFunc: func(old, obj interface{}) {
-			utils.AviLog.Infof("NSX Network Config Update Event")
+			utils.AviLog.Infof("Namespace Update Event")
 		},
 		DeleteFunc: func(obj interface{}) {
-			utils.AviLog.Infof("NSX Network Config Delete Event")
+			utils.AviLog.Infof("Namespace Delete Event")
 			crd := obj.(*unstructured.Unstructured)
 			_, found, err := unstructured.NestedStringMap(crd.UnstructuredContent(), "spec")
 			if err != nil || !found {
-				utils.AviLog.Warnf("nsxnetworkconfiguration spec not found: %+v", err)
+				utils.AviLog.Warnf("Namespace spec not found: %+v", err)
 				return
 			}
-			handleNSXNetworkConfigDelete()
+			c.handleNamespaceDelete()
 		},
 	}
-	c.dynamicInformers.NSXNetworkConfigInformer.Informer().AddEventHandler(nsxNetworkConfigHandler)
+	c.informers.NSInformer.Informer().AddEventHandler(namespaceHandler)
 
-	go c.dynamicInformers.NSXNetworkConfigInformer.Informer().Run(stopCh)
-	if !cache.WaitForCacheSync(stopCh, c.dynamicInformers.NSXNetworkConfigInformer.Informer().HasSynced) {
+	go c.informers.NSInformer.Informer().Run(stopCh)
+	if !cache.WaitForCacheSync(stopCh, c.informers.NSInformer.Informer().HasSynced) {
 		runtime.HandleError(fmt.Errorf("Timed out waiting for caches to sync"))
 	} else {
-		utils.AviLog.Info("Caches synced for nsxnetworkconfiguration informer")
+		utils.AviLog.Info("Caches synced for Namespace informer")
 	}
 }
 
-func addNSXNetworkConfigCount() error {
-	dynamicClient := lib.GetVCFDynamicClientSet()
-	crdClient := dynamicClient.Resource(lib.NSXNetworkConfigGVR)
-	crdList, err := crdClient.List(context.TODO(), metav1.ListOptions{})
+func (c *VCFK8sController) addWorkloadNamespaceCount() error {
+	nsList, err := c.informers.NSInformer.Lister().List(labels.Set(nil).AsSelector())
```

Comment:
should we add code from 176- 179 in getWorkloadNamespaceCount() function. These lines are repeated in handleNamespaceAdd(), handleNamespaceDelete().